File: src/main/java/de/thatsich/common/module/block/ABlock.java
package de.thatsich.common.module.block;

import de.thatsich.common.module.block.info.ABlockInfo;
import de.thatsich.common.module.block.info.ABlockName;
import net.minecraft.block.Block;
import net.minecraft.creativetab.CreativeTabs;


public abstract class ABlock extends Block
{
	protected ABlock ( ABlockInfo info, ABlockConfig config, ABlockTexture blockTexture )
	{
		super( info.getMaterial() );

		final ABlockName blockName = info.getBlockName();
		final String name = blockName.getName();
		final float hardness = info.getHardness();
		final CreativeTabs creativeTab = info.getCreativeTab();
		final String texture = blockTexture.getTexture();

		this.setBlockName( name );
		this.setHardness( hardness );
		this.setBlockTextureName( texture );
		//		this.setBlockBounds(  );

		this.setCreativeTab( creativeTab );
	}
}

File: src/main/scala/de/thatsich/minecraft/core/module/block/ABlock.scala
package de.thatsich.minecraft.core.module.block

import net.minecraft.block.Block
import net.minecraft.creativetab.CreativeTabs

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class ABlock( info: ABlockInfo, config: ABlockConfig )
	extends Block( info.getMaterial )
	        with IBlock
{
	val name       : String       = info.getName
	val hardness   : Float        = info.getHardness
	val creativeTab: CreativeTabs = info.getCreativeTab
	val texture    : String       = info.getTexture

	this.setBlockName( name )
	this.setHardness( hardness )
	this.setBlockTextureName( texture )
	this.setCreativeTab( creativeTab )
}
